presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java

```diff
@@ -110,13 +109,20 @@ private Optimizer(PlanNodeIdAllocator idAllocator, SymbolAllocator symbolAllocat
         @Override
         public PlanNode visitAggregation(AggregationNode node, RewriteContext<Optional<AggregateInfo>> context)
         {
-            // optimize if and only if
-            // some aggregation functions have a distinct mask symbol
-            // and if not all aggregation functions on same distinct mask symbol (this case handled by SingleDistinctOptimizer)
             List<Symbol> masks = node.getAggregations().values().stream()
                     .map(Aggregation::getMask).filter(Optional::isPresent).map(Optional::get).collect(toImmutableList());
-            Set<Symbol> uniqueMasks = ImmutableSet.copyOf(masks);
-            if (uniqueMasks.size() != 1 || masks.size() == node.getAggregations().size()) {
+
+            if (masks.size() == 0 || !masks.stream().allMatch(mask -> mask.getName().endsWith("$distinct"))) {
```

Comment:
This is not a very stable way to tell whether the mask is a `DISTINCT`. What would happen if some of the masks are `FILTER` etc? Technically this rule can still apply right?